packages/block-library/src/media-text/edit.js

```diff
@@ -243,7 +243,7 @@ function MediaTextEdit( {
 		'is-selected': isSelected,
 		'is-stacked-on-mobile': isStackedOnMobile,
 		[ `is-vertically-aligned-${ verticalAlignment }` ]: verticalAlignment,
-		'is-image-fill': imageFill,
+		'is-image-fill-v8': imageFill,
```

Comment:
I'm not sure we can avoid the new class name in order to keep back compat for older versions. @Mamaduka do you see another way around for this? 🤔 

If we can't avoid it let's find a better name without the version number.

Reply:
Perhaps something that references the new mechanism we're using for these? Here are some ideas:
- `is-image-fill-no-bg`
- `is-image-fill-object`
- `is-image-fill-element`

Alternatively, we could align more with the wording in the UI ("Crop image to fill"), so:
- `is-image-crop-to-fill`
- `is-image-cropped`

Reply:
As for why I opted for a new classname in the first place, let me explain in detail.

The issue is that we need new styles because of the markup changes, and these styles are incompatible with the previous ones because of the nature of these changes (changing display behaviour for the `<img>` element).

This leaves us with three options, as I understand:

### Replace the classname
The idea here is to create a new classname for the "Crop image to fill" option, to replace the previous one.

Markup example:

```
<div class="wp-block-media-text is-image-fill-v8">
```

This is the option I went with in this PR.

### Add another classname

The idea here is to keep the old classname for the "Crop image to fill" option, but add a new one to indicate the new markup is being used.

Markup example:

```
<div class="wp-block-media-text is-image-fill is-v8">
```

I opted against this approach, since it would have meant additional CSS to return the properties changed in `is-image-fill` to their default values, and it didn't seem to come with significant benefits.

### Upgrade the markup

Another option would be to treat the block as dynamic, and rewrite its content in PHP if the "Crop image to fill" option is enabled. This would ensure the new markup is always used, removing the need to preserve backwards compatibility with styles.

Currently, this block is only dynamic when using a featured image. I opted against this approach because it would have meant further usage of dynamic blocks, which can result in increased server processing times.

### Other approaches?

Are there any other approaches I'm not aware of that we should be considering?

Reply:
Unless we get feedback for a possible different approach, the new class name seems fine to me. Maybe the name could be something like `is-image-fill-element` you suggested, but this is the easiest change we can do later, before merging.